File: python/leetcode/257.py
class Solution:
    def binaryTreePaths(self, root):
        res = []
        self.dfs(root, "", res)
        return res

    def dfs(self, root, ans, res):
        if not root: return 
        if not root.left and not root.right:
            res.append(ans + str(root.val))
            return
        if root.left:
            self.dfs(root.left, ans + str(root.val) + "->", res)
        if root.right:
            self.dfs(root.right, ans + str(root.val) + "->", res)

File: cpp/leetcode/257.cpp
class Solution {
public:
    vector<string> binaryTreePaths(TreeNode* root) {
        if(!root) return {};
        vector<string> res;
        help(root, "", res);
        return res;
    }
    
    void help(TreeNode* root, string ans, vector<string> & res){
        if(!root) return;
        if(!root->left && !root->right){
            res.push_back(ans + to_string(root->val));
            return;
        }
        if(root->left) help(root->left, ans + to_string(root->val)+ "->", res);
        if(root->right) help(root->right, ans + to_string(root->val) + "->", res);
    }
};